### PR TITLE
Queen Hiveweed Building

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

This makes it so that queens can only build thick structures on hiveweeds, like when she needs to build up the hive during the starting stages of the game, or protecting areas around hive clusters.

Queens can build thick structures on normal weeds while her building speed is boosted (the first 30 minutes after the round starts).

## Why It's Good For The Game

Queens being able to remote build thick resin structures directly at FOB was a bit nonsense, at least now you need to create a hive cluster there, which gives the marines a small objective to destroy.

## Changelog

:cl:
balance: Ovi'd queens can now only build thick walls/doors/membrane on hiveweeds, otherwise it'll just be normal. Building them costs an amount of plasma roughly in the middle of the two. The queen can build thick structures on normal weeds while her building is boosted (first 30 minutes of the round).
balance: Ovi'd queens can no longer be fed plasma by drones.
/:cl:

### CODE REVIEW

**Comment by Carnage068** [comment]
Why are we taking plasma feeding away from the Queen, again? If a drone wants to sit at hive and help out the queen it should be able to do that.

**Comment by Git-Nivrak** [comment]
Queen builds hive cluster on frontline -> Spams thick resin walls -> Same problem.
IMO you should make it so it's only thick within 35 tiles of the Queen / Hivecore

**Review by stanalbatross — COMMENTED** [review]
This is a massive nerf to ovi building at all points in the round especially in the crucial predrop stage when the queen is vital to making good hive defences. In my opinion it's too much of a nerf for an easily avoidable and counter able thing especially since we removed it going under Cades. DNM for further discussion.

**Comment by Chainsawicus** [comment]
> balance: Ovi'd queens can no longer be fed plasma by drones.

Remove this from the PR and it's a perfect change IMO.

**Comment by PalJohnson** [comment]
Honestly, this is a really good chance. The only other alternative to consider is perhaps making thick walls longer to make in order to make it less spammy and encourage caste like the **_hivelord_** to be a lot more valuable as they can upgrade resin walls to thick walls. There are ways to continually build thick walls without the need of the queen using it via thick eye. No one really plays hive lord atm as they get outclassed by a longshot by the queen eye. Sure you can still build them but not as fast as before allowing for more time to be allocated to at least try to counter it. Being able to produce them so fast was a bit nuts, especially with flamer weapons not being readily available for it. I know as a survivor I've seen queen block people off with thick weeds in the thick of fighting or just stonewall holdouts so that Xenos could pickoff select areas of barricades with survivors not being able to counter them due to lack of flame or AP weapons.

**Comment by PalJohnson** [comment]
> This is a massive nerf to ovi building at all points in the round especially in the crucial predrop stage when the queen is vital to making good hive defences. In my opinion it's too much of a nerf for an easily avoidable and counter able thing especially since we removed it going under Cades. DNM for further discussion.

Unless you have a flame weapon then countering the heavy wall spam can be rather difficult to counter especially when they are constructed as you would need a machete or master key to take them down quickly. Usually if you are on a barricade you are shooting constantly, especially during fob defense. In the case of the attacks, you can destroy weeds quicker but if you look away for a second then the walls begin to appear which really isn't fun to play against. At least its not total removal as hive clusters would still be needed. Hivelordss would be a little more valuable as they can upgrade walls.

File: code/modules/mob/living/carbon/xenomorph/castes/Queen.dm [hunk 1]
```diff
@@ -855,3 +855,9 @@
 
 /mob/living/carbon/Xenomorph/Queen/gib(var/cause = "gibbing")
 	death(cause, 1)
+
+/mob/living/carbon/Xenomorph/Queen/override_secrete_thick_resin()
+	var/datum/action/xeno_action/activable/secrete_resin/remote/queen/resin_ability = get_xeno_action_by_type(src, /datum/action/xeno_action/activable/secrete_resin/remote/queen)
+	if(resin_ability?.boosted)
+		return TRUE
+	return FALSE
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/castes/Queen.dm:863]: make this a signal instead

File: code/modules/mob/living/carbon/xenomorph/resin_constructions.dm [hunk 2]
```diff
@@ -63,15 +66,35 @@ GLOBAL_VAR_INIT(resin_lz_allowed, FALSE)
...
 // Subtype encompassing all resin constructions that are of type /turf
-/datum/resin_construction/resin_turf/build(var/turf/T, var/hivenumber, var/builder)
-	T.PlaceOnTop(build_path)
+/datum/resin_construction/resin_turf/build(var/turf/T, var/hivenumber, var/mob/living/carbon/Xenomorph/builder)
+	var/path = build_path
+	var/can_build_thick = TRUE
+	if(thick_hiveweed)
+		var/obj/effect/alien/weeds/W = locate() in T
+		if(!W || W.hivenumber != hivenumber || W.weed_strength < WEED_LEVEL_HIVE)
+			can_build_thick = FALSE
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/resin_constructions.dm:92]: W needs a more descriptive variable name. The code is also duplicated and T also needs a more descriptive variable name

File: code/modules/mob/living/carbon/xenomorph/resin_constructions.dm [hunk 3]
```diff
@@ -63,15 +66,35 @@ GLOBAL_VAR_INIT(resin_lz_allowed, FALSE)
...
+
+	if(build_path_thick && (can_build_thick || builder.override_secrete_thick_resin()))
+		path = build_path_thick
+
+	if(pass_hivenumber)
+		return new path(T, hivenumber, builder)
+	return new path(T)
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/resin_constructions.dm:82]: T needs a more descriptive variable name and so does W

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities/ability_helper_procs.dm [hunk 4]
```diff
@@ -410,6 +410,12 @@
 		to_chat(src, SPAN_WARNING("You need to be closer to [target]."))
 		return
 
+	if(isXenoQueen(target))
+		var/mob/living/carbon/Xenomorph/Queen/Q = target
+		if(Q.ovipositor)
+			to_chat(src, SPAN_WARNING("You can't transfer plasma to a queen mounted on her ovipositor."))
+			return
```
[the remark this answers is not in the document]
Comment (Watermelon914) [reply code/modules/mob/living/carbon/xenomorph/abilities/ability_helper_procs.dm:417]: Q also needs a more descriptive variable name

**Pull request merged**